### PR TITLE
ENH: Create 2DTransfrom.py

### PR DESCRIPTION
**Pull request by Yukti-09**
Picking up from 

https://github.com/numpy/numpy/pull/17089  
https://github.com/scipy/scipy/issues/12693
https://github.com/numpy/numpy/issues/17043

I have added transformations in 2D.

### CODE REVIEW

File: scipy/spatial/transform/2DTransfrom.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+def rotation(angle):
+	arr = np.empty([2, 2])
+
+	arr[0][0] = math.cos(angle)
+	arr[0][1] = -math.sin(angle)
+	arr[1][0] = math.sin(angle)
+	arr[1][1] = math.cos(angle)
+
+	return arr #This returns a rotation matrix
```
Comment (eric-wieser) [inline scipy/spatial/transform/2DTransfrom.py:18]: If this is going to be worth including (anywhere) in scipy, it's probably worth makign it broadcasting:
```python
c = np.cos(angle)
s = np.sin(angle)
arr = np.empty(c.shape + (2, 2), dtype=c.dtype)
arr[..., 0, 0] = c
arr[..., 0, 1] = -s
arr[..., 1, 0] = s
arr[..., 1, 1] = c
```

Comment (Yukti-09) [reply scipy/spatial/transform/2DTransfrom.py]: I have made the required changes.

**Comment by tylerjereddy** [comment]
Before making more detailed suggestions, I'm curious what @nmayorov thinks about the 2D transformations idea given all their work on the 3D/quaternion stuff in SciPy.

**Comment by Yukti-09** [comment]
@tylerjereddy 
I have raised an issue here as well:
https://github.com/scipy/scipy/issues/12693

**Comment by nmayorov** [comment]
> Before making more detailed suggestions, I'm curious what @nmayorov thinks about the 2D transformations idea given all their work on the 3D/quaternion stuff in SciPy.

Well, 2D rotations are much more simple and universally described by a single angle. Can be seen as a 3D rotations around Z (for example) axis.

Providing `Rotation2D` class seems like a mismatch between API complexity and simplicity of the subject. I don't think its sane.

As for the provided simple implementation -- I guess I'm about neutral on its inclusion (after implementation and docstring cleanups).

There are some discussions about providing rotations in N-dimensions in a selected plane. I think it's not often used and too abstract (besides linear algebra maybe, but they definitely don't want to use scipy in their compiled code) As for rotation as part of 4D transformation matrix in homogeneous coordinates -- maybe we could go with `Transform` class (combining `Rotation` + translation), which can be represented as 4D matrix if desired.

**Comment by Yukti-09** [comment]
@nmayorov 
I have created an issue for the same https://github.com/scipy/scipy/issues/12693
2D rotation matrices are used very frequently in linear algebra. It would be great if one could give the angle as input and get a rotation matrix.

**Pull request closed without merging**